File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file242.f90
subroutine init_array(ni, nj, nk, alpha, beta, c, a, b)
implicit none

DATA_TYPE, dimension(nk, ni) :: a
DATA_TYPE, dimension(nj, nk) :: b
DATA_TYPE, dimension(nj, ni) :: c
DATA_TYPE :: alpha, beta
integer :: ni, nj, nk
integer :: i, j

alpha = 32412
beta = 2123

do i = 1, ni
do j = 1, nj
c(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(ni)
end do
end do
do i = 1, ni
do j = 1, nk
a(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(ni)
end do
end do
do i = 1, nk
do j = 1, nj
b(j, i) = ((DBLE(i - 1) * DBLE(j - 1))) / DBLE(ni)
end do
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file242.cpp
void init_array(int ni, int nj, int nk, double alpha, double beta, double c[nj][ni], double a[nk][ni], double b[nj][nk]){
 alpha = 32412;
 beta = 2123;
 for (int i = 0; i < ni; i++){
 for (int j = 0; j < nj; j++){
 c[j][i] = ((double(i - 1) * double(j - 1))) / double(ni);
 }
 }
 for (int i = 0; i < ni; i++){
 for (int j = 0; j < nk; j++){
 a[j][i] = ((double(i - 1) * double(j - 1))) / double(ni);
 }
 }
 for (int i = 0; i < nk; i++){
 for (int j = 0; j < nj; j++){
 b[j][i] = ((double(i - 1) * double(j - 1))) / double(ni);
 }
 }
}
